reference/multigrid/amgx_pgm_kernels.cpp

```diff
@@ -231,6 +296,59 @@ GKO_INSTANTIATE_FOR_EACH_NON_COMPLEX_VALUE_AND_INDEX_TYPE(
     GKO_DECLARE_AMGX_PGM_ASSIGN_TO_EXIST_AGG);
 
 
+template <typename ValueType, typename IndexType>
+void sort_row_major(std::shared_ptr<const DefaultExecutor> exec, size_type nnz,
+                    IndexType* row_idxs, IndexType* col_idxs, ValueType* vals)
+{
+    auto it = detail::make_zip_iterator(row_idxs, col_idxs, vals);
+    std::stable_sort(it, it + nnz, [](auto a, auto b) {
```

Comment:
Is there a particular reason you need `stable_sort` ?

Reply:
when the reduce is determinstic, it will also require the element position determinstic.

Reply:
In each row, each value has a unique col_idx right ? If the the col_idx is the same, you sum them up later anyway, right ?

Reply:
no, some value might have the same col_idx. it is used after fine matrix -> coarse matrix index mapping.